lang/src/org/partiql/lang/eval/ExprValueFactory.kt

```diff
@@ -205,6 +220,34 @@ private class ExprValueFactoryImpl(override val ion: IonSystem) : ExprValueFacto
     override fun newTimestamp(value: Timestamp): ExprValue =
         TimestampExprValue(ion, value)
 
+    override fun newTime(value: LocalTime, zoneOffset: ZoneOffset?): ExprValue {
+        return TimeExprValue(ion, value, zoneOffset)
+    }
+
+    override fun newTime(hour: Int, minute: Int, second: Int, nano: Int, precision: Int?, tz_minutes: Int?): ExprValue {
+        // Round nanoseconds to the given precision.
+        val nanoWithPrecision = when  {
+            precision == null || precision >= 9 -> nano
```

Comment:
A default value for a non-nullable `precision` parameter avoids the need for the null check here.

Reply:
Also need to consider `precision < 0`, I think, even if the parser rejects the negative sign.

Reply:
I see. Added the check for the `precision`.